### PR TITLE
Feature/michael/add events

### PR DESCRIPTION
**Pull request by Michael-B-D**
after correcting some notes

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (yammesicka) [inline .vscode/settings.json:1]: Be sure this file is in the `.gitignore`.
Remove it using `git rm` if needed.

File: app/database/config.py [hunk 2]
```diff
@@ -0,0 +1 @@
+db_url = "sqlite:///./calendar_app.db"
```
Comment (yammesicka) [inline app/database/config.py:1]: Please use `config.py.example`

File: app/database/database.py [hunk 3]
```diff
@@ -5,7 +5,10 @@
 from sqlalchemy.orm import sessionmaker
 
 
-SQLALCHEMY_DATABASE_URL = os.getenv("DATABASE_CONNECTION_STRING")
+from .config import db_url
```
Comment (yammesicka) [inline app/database/database.py:8]: Please import it as an absolute import

File: app/database/models.py [hunk 4]
```diff
@@ -22,8 +22,23 @@ class Event(Base):
...
     id = Column(Integer, primary_key=True, index=True)
     title = Column(String)
+    start_date = Column(DateTime)
+    end_date = Column(DateTime)
+    vc_link = Column(String)
     content = Column(String)
-    date = Column(DateTime)
+    location = Column(String)
     owner_id = Column(Integer, ForeignKey("users.id"))
+    # repeated_event_id = Column(Integer, ForeignKey("repeated_events.id"))
```
Comment (yammesicka) [inline app/database/models.py:31]: Please remove commented out code

File: app/main.py [hunk 5]
```diff
@@ -1,18 +1,38 @@
...
 
+from app.database import models
+from app.database.models import Event
+from app.database.database import SessionLocal, engine
+
+models.Base.metadata.create_all(bind=engine)
+
+
+
+def get_db():
```
Comment (yammesicka) [inline app/main.py:18]: Already exists in another file :)

File: app/main.py [hunk 6]
```diff
@@ -30,3 +50,60 @@ def profile(request: Request):
         "username": current_username,
         "events": upcouming_events
     })
+
+
+@app.get("/profile/{user_id}/EditEvent", response_class=HTMLResponse)
```
Comment (yammesicka) [inline app/main.py:55]: Prefer adding this endpoint on the relevant file on app/routers. The `main.py` file should be clean.

File: app/templates/profile.html [hunk 7]
```diff
@@ -9,7 +9,6 @@
             <div class="col-3">
                 <!-- Profile card -->
                 <div class="card card-profile mb-3">
```
Comment (yammesicka) [inline app/templates/profile.html:11]: Why have you removed this line?

File: app/templates/profile.html [hunk 8]
```diff
@@ -9,7 +9,6 @@
             <div class="col-3">
```
Comment (yammesicka) [inline app/templates/profile.html:9]: Please remove the database file

File: tests/test_app.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    assert response.status_code == 200
+
+
+def test_session_db():
+    assert get_db() is not None
+
+
+def test_get_profile():
+    response = client.get("/profile/123/EditEvent")
+    assert response.status_code == 200
```
Comment (yammesicka) [inline tests/test_app.py:34]: Prefer `.ok`

File: tests/test_app.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+from fastapi import FastAPI
+from fastapi.testclient import TestClient
+import pytest
+
+from app.main import add_event, app, check_validation, get_db
+
+client = TestClient(app)
+date_test_data = [(datetime.today() - timedelta(1), datetime.today())]
+event_test_data = [(
```
Comment (yammesicka) [inline tests/test_app.py:11]: Can you please add this event as a fixture?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 11]
```diff
@@ -30,3 +50,60 @@ def profile(request: Request):
...
+def check_validation(start_time, end_time) -> bool:
+    """Check if the start_date is smaller then the end_time"""
+    try:
+        return start_time < end_time
+    except TypeError:
+        return False
+
+
+
+def add_event(values: dict, db) -> Event:
```
Comment (ivarshav) [inline app/main.py:96]: Move to `internal/event` ?

File: app/main.py [hunk 12]
```diff
@@ -1,18 +1,38 @@
...
+        db = SessionLocal()
+        yield db
+    finally:
+        db.close()
+
 
 app = FastAPI()
 
-app.mount("/static", StaticFiles(directory="static"), name="static")
+# app.mount("static", StaticFiles(directory="static"), name="static")
```
Comment (ivarshav) [inline app/main.py]: Remove this line? :)

**Comment by yammesicka** [comment]
Can you please send this PR to `:develop` and not to `:main`? :)

**Comment by Michael-B-D** [comment]
make a new PR after sync with the ripo updates

**Pull request closed without merging**